src/main/java/seedu/duke/command/cap/CapCommand.java

```diff
@@ -62,23 +88,47 @@ private double gradeConvert(String grade) {
         return score;
     }
 
-    private double calculateCap() {
+    /**
+     * Calculates the cap of the user.
+     *
+     * @return
+     * cap of the user
+     * @throws InvalidCapAttained
+     * When the attained cap is unimaginable
+     */
+    private double calculateCap() throws InvalidCapAttained {
         HashMap<String, Module> moduleList = ModuleManager.getModulesMap();
```

Comment:
As mentioned before, please use `ModuleManager.getModCodeList()` instead of directly accessing the `modulesMap` object.